app/api/server/v1/roles.js

```diff
@@ -104,3 +123,53 @@ API.v1.addRoute('roles.getUsersInRole', { authRequired: true }, {
 		return API.v1.success({ users: users.fetch(), total: users.count() });
 	},
 });
+
+API.v1.addRoute('roles.update', { authRequired: true }, {
+	post() {
+		check(this.bodyParams, {
+			roleId: String,
+			name: Match.Maybe(String),
+			scope: Match.Maybe(String),
+			description: Match.Maybe(String),
+		});
+
+		const roleData = {
```

Comment:
no option to update the `mandatory2fa` attribute?

Reply:
I thought the `mandatory2FA` should not be updated by the user via the API as the `roles.create` also doesn't have this parameter, my bad! I'll fix it, thanks for noticing.